gym/wrappers/compatibility.py

```diff
@@ -0,0 +1,130 @@
+"""A compatibility wrapper converting an old-style environment into a valid environment."""
+import sys
+from typing import Any, Dict, Optional, Tuple
+
+import gym
+from gym.core import ObsType
+from gym.utils.step_api_compatibility import convert_to_terminated_truncated_step_api
+
+if sys.version_info >= (3, 8):
+    from typing import Protocol, runtime_checkable
+elif sys.version_info >= (3, 7):
+    from typing_extensions import Protocol, runtime_checkable
```

Comment:
`typing_extension` does the version check internally, is it necessary to check here? pytorch seems to simply use `typing_extensions`.

Reply:
Your right but this shouldn't change the implementation